clouddriver-aws/src/main/java/com/netflix/spinnaker/clouddriver/aws/services/LaunchTemplateService.java

```diff
@@ -247,6 +247,10 @@ private RequestLaunchTemplateData buildLaunchTemplateData(
     networkInterfaceRequest.setDeviceIndex(0);
     networkInterfaceRequest.setGroups(description.getSecurityGroups());
 
+    // Nitro Enclave options
+    request.setEnclaveOptions(
```

Comment:
Instead of setting this property with every create launch template request, it might make sense to set enclave options only when specified (i.e. the default in the description will be null)?
Thinking of (current / future) features that might be incompatible with it. One such feature is `hibernationOptions` which is not currently supported in Spinnaker, but just throwing it out there for forward compatibility :)

Reply:
@iandelahorne `associateIPv6Address` and  `unlimitedCpuCredits` are two launch template only fields that have a null default. Note that there might some extra checks for `unlimitedCpuCredits` in other files, which will not be necessary for `enableEnclave`.

For the change I requested, wrapping the `request.setEnclaveOptions` in a null check like on L#235 or L#353 should do.